### PR TITLE
refactor: use `AtomicOp` in `LoweredAst` pt. 3

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/6161

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala [hunk 1]
```diff
@@ -245,6 +176,50 @@ object Simplifier {
         SimplifiedAst.JvmMethod(ident, fparams, exp, retTpe, simplifyEffect(eff), loc)
     }
 
+    /**
+      * Translates the given `ApplyAtomic` expression to the corresponding [[SimplifiedAst]] expression.
+      */
+    def visitAtomic(atomicExp: LoweredAst.Expression.ApplyAtomic): SimplifiedAst.Expression.ApplyAtomic = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala:182]: We usually don't do this. Can we inline this instead?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala:182]: Its a function thats only called once, from one place.

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala:182]: Sure :)

**Review by magnus-madsen — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala [hunk 2]
```diff
@@ -93,6 +90,26 @@ object Simplifier {
...
+
+          case AtomicOp.Lazy =>
+            // Wrap the expression in a closure: () -> tpe \ Pure
+            val e = es.head
+            val lambdaTyp = Type.mkArrowWithEffect(Type.Unit, Type.Pure, e.tpe, loc)
+            val fp = SimplifiedAst.FormalParam(Symbol.freshVarSym("_lazy", BoundBy.FormalParam, loc), Ast.Modifiers.Empty, Type.mkUnit(loc), loc)
+            val lambdaExp = SimplifiedAst.Expression.Lambda(List(fp), e, lambdaTyp, loc)
+            SimplifiedAst.Expression.ApplyAtomic(AtomicOp.Lazy, List(lambdaExp), tpe, Purity.Pure, loc)
+
+          case AtomicOp.HoleError(_) =>
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala:109]: What about MatchError then?

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala:109]: That's created in Simplifier so it's not a problem :) https://github.com/flix/flix/blob/f63c7e1ed70856545cf42e7bbd2143c792d0ecd0/main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala#L380

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala:109]: Again, its the brain that is the problem!

**Pull request merged**